resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/StreamingCompatibilitySuite.scala

```diff
@@ -0,0 +1,220 @@
+        appArgs = Array[String](host, port.toString),
+        expectedJVMValue = Seq("spark-streaming-kube"))
+    } finally {
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark structured streaming in cluster mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    try {
+      runSparkJVMCheckAndVerifyCompletion(
+        mainClass = "org.apache.spark.examples.sql.streaming.StructuredNetworkWordCount",
+        appArgs = Array[String](host, port.toString),
+        expectedJVMValue = Seq("spark-streaming-kube"))
+    } finally {
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark structured streaming in client mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    val (driverPort: Int, blockManagerPort: Int, driverService: Service) =
+      driverServiceSetup(driverPodName)
+    try {
+      val driverPod = setupSparkStreamingPod("client", driverPodName)
+        .addToArgs("--conf",
+          s"spark.driver.host=" +
+            s"${driverService.getMetadata.getName}.${kubernetesTestComponents.namespace}.svc")
+        .addToArgs("--conf", s"spark.driver.port=$driverPort")
+        .addToArgs("--conf", s"spark.driver.blockManager.port=$blockManagerPort")
+        .addToArgs("sql.streaming.StructuredNetworkWordCount")
+        .addToArgs(host, port.toString)
+        .endContainer()
+        .endSpec()
+        .done()
+
+      val TIMEOUT = PatienceConfiguration.Timeout(Span(3, Minutes))
+      Eventually.eventually(TIMEOUT, INTERVAL) {
+        assert(getRunLog(driverPodName).contains("spark-streaming-kube"),
+          "The application did not complete.")
+      }
+    }
+    finally {
+      // Have to delete the service manually since it doesn't have an owner reference
+      kubernetesTestComponents
+        .kubernetesClient
+        .services()
+        .inNamespace(kubernetesTestComponents.namespace)
+        .delete(driverService)
+      serverSocket.close()
+    }
+  }
+
+  private def getRunLog(_driverPodName: String): String = kubernetesTestComponents.kubernetesClient
+    .pods()
+    .withName(_driverPodName)
+    .getLog
+
+  private def setupSparkStreamingPod(mode: String, _driverPodName: String) = {
+    val labels = Map("spark-app-selector" -> _driverPodName)
+    testBackend
+      .getKubernetesClient
+      .pods()
+      .inNamespace(kubernetesTestComponents.namespace)
+      .createNew()
+      .withNewMetadata()
+      .withName(_driverPodName)
+      .withLabels(labels.asJava)
+      .endMetadata()
+      .withNewSpec()
+      .withServiceAccountName(kubernetesTestComponents.serviceAccountName)
+      .addNewContainer()
+      .withName("spark-example")
+      .withImage(image)
+      .withImagePullPolicy("IfNotPresent")
+      .withCommand("/opt/spark/bin/run-example")
+      .addToArgs("--master", s"k8s://https://kubernetes.default.svc")
+      .addToArgs("--deploy-mode", mode)
```

Comment:
If we're creating the pod manually, the deploy mode should always be client and not cluster, so this should just be hardcoded, and the argument removed from the function signature.

Reply:
Also why are we running in client mode if we're going to be deploying a pod anyways? We can deploy this in cluster mode using spark submit and then set up the service in front of the pod anyways.